index.js

```diff
@@ -1,34 +1,39 @@
 'use strict';
 
-// Key in annotations object that maps to an executor NPM module
-// value is a placeholder until resolution to screwdriver-cd/screwdriver#595
-const ANNOTATION_EXECUTOR_TYPE = 'beta.screwdriver.cd/executor';
+const hoek = require('hoek');
+const ANNOTATION_EXECUTOR_TYPE = 'beta.screwdriver.cd/executor'; // Key in annotations object that maps to an executor NPM module
 const Executor = require('screwdriver-executor-base');
 
 class ExecutorRouter extends Executor {
     /**
      * Constructs a router for different Executor strategies.
      * @method constructor
-     * @param  {Object}         options
-     * @param  {Array|String}   options.plugins                Array of plugins to load or a single plugin string
-     * @param  {String}         options.plugins[x].moduleName  Name of the executor NPM module to load
-     * @param  {String}         options.plugins[x].options     Configuration to construct the module with
+     * @param  {Object}         config                      Object with executor and ecosystem
+     * @param  {Object}         config.ecosystem            Object with ecosystem values
+     * @param  {Array|Object}   config.executor             Array of executors to load or a single executor object
+     * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
+     * @param  {String}         config.executor[x].options  Configuration to construct the module with
      */
-    constructor(options = {}) {
+    constructor(config) {
+        if (!config) {
+            throw new Error('No executor config passed in.');
+        }
+
         super();
 
-        const plugins = options.plugins;
-        const registrations = (Array.isArray(plugins)) ? plugins : [plugins];
+        const ecosystem = config.ecosystem || {};
+        const executorConfig = config.executor;
+        const registrations = (Array.isArray(executorConfig)) ? executorConfig : [executorConfig];
 
         registrations.forEach((plugin, index) => {
             // eslint-disable-next-line global-require, import/no-dynamic-require
             const ExecutorPlugin = require(plugin.name);
-
-            const executorPlugin = new ExecutorPlugin(plugin.options);
+            const options = hoek.applyToDefaults({ ecosystem }, plugin.options); // Add ecosystem to executor options
```

Comment:
`applyToDefaults` is can also be done without having to `include` hoek with [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)